nifi-nar-bundles/nifi-py4j-bundle/nifi-py4j-bridge/src/main/java/org/apache/nifi/py4j/PythonProcess.java

```diff
@@ -267,6 +264,21 @@ private Process launchPythonProcess(final int listeningPort, final String authTo
         return processBuilder.start();
     }
 
+    String resolvePythonCommand() throws IOException {
+        final File pythonCmdFile = new File(processConfig.getPythonCommand());
+        final String pythonCmd = pythonCmdFile.getName();
+        File[] virtualEnvFileList = virtualEnvHome.listFiles((file, name) -> file.isDirectory() && (name.equals("bin") || name.equals("Scripts")));
+        //Prefer bin directory
+        String commandFileExecutableDirectory = "bin";
+        if(virtualEnvFileList == null || virtualEnvFileList.length == 0) {
+            throw new IOException("Python binary directory could not be found in " + virtualEnvHome);
+        }else if( virtualEnvFileList.length == 1) {
+            commandFileExecutableDirectory = virtualEnvFileList[0].getName();
+        }
```

Comment:
Instead of declaring and resetting the value, recommend declaring the directory as final and setting it once. Also note spacing adjustments:
```suggestion
        final String commandBinaryDirectory;
        if (virtualEnvFileList == null || virtualEnvFileList.length == 0) {
            throw new IOException("Python virtual environment binary directory could not be found in " + virtualEnvHome);
        } else if (virtualEnvFileList.length == 1) {
            commandBinaryDirectory = virtualEnvFileList[0].getName();
        } else {
            // Default to bin as the default for Linux and macOS
            commandBinaryDirectory = "bin";
        }
```